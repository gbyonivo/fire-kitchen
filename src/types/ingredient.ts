export interface SampleIngredient {
  idIngredient?: string;
  strIngredient?: string;
  strDescription?: string;
  strType?: string;
}

export function isSampleIngredient(
  ingredient: unknown
): ingredient is SampleIngredient {
  return (
    typeof ingredient === "object" &&
    ingredient !== null &&
    "idIngredient" in ingredient &&
    "strIngredient" in ingredient
  );
}

export interface Ingredient {
  id: string;
  ingredient: string;
  description?: string;
  type?: string;
}
