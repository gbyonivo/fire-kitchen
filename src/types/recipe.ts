export interface RecipeIngredient {
  name: string;
  measure: string;
}

export interface Recipe {
  id: string;
  name: string;
  category: string;
  area: string;
  instructions: string;
  thumbnail: string;
  tags: string[];
  youtube: string;
  ingredients: RecipeIngredient[];
}
