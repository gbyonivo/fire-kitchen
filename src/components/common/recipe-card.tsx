import { Recipe } from "@/types/recipe";
import Image from "next/image";
interface RecipeCardProps {
  recipe: Recipe;
  id?: string;
}

export function RecipeCard({ recipe, id }: RecipeCardProps) {
  return (
    <div
      className="bg-black rounded-xl border border-white/10 flex cursor-pointer 
    flex-col transition-all duration-300 hover:scale-105"
      data-testid={id}
    >
      <Image
        src={recipe.thumbnail}
        alt={recipe.name}
        width={150}
        height={150}
        className="w-full h-fullobject-cover rounded-t-xl"
      />
      <div className="p-2 h-20 overflow-auto">
        <h2 className="text-white text-sm font-bold">{recipe.name}</h2>
        <p className="text-white/50 text-sm">{recipe.category}</p>
        <p className="text-white/50 text-sm">{recipe.area}</p>
      </div>
    </div>
  );
}
