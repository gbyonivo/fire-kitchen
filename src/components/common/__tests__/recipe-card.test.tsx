import { render } from "@testing-library/react";
import { RecipeCard } from "../recipe-card";
import { Recipe } from "@/types/recipe";

const recipe: Recipe = {
  id: "1",
  name: "Test",
  ingredients: [],
  category: "Test",
  area: "Test",
  instructions: "Test",
  thumbnail: "https://example.com/thumbnail.jpg",
  tags: [],
  youtube: "https://example.com/youtube.com",
};

describe("Spinner", () => {
  test.each([
    [recipe],
    // should truncate long text
    [{ ...recipe, name: "longtextlongtextlongtextlongtextlongtextlongtext" }],
  ])("%s should match snapshot", (recipe) => {
    const { container } = render(<RecipeCard recipe={recipe} />);
    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
