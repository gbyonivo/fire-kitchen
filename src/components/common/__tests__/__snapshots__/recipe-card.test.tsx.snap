// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner {
  id: '1',
  name: 'Test',
  ingredients: [],
  category: 'Test',
  area: 'Test',
  instructions: 'Test',
  thumbnail: 'https://example.com/thumbnail.jpg',
  tags: [],
  youtube: 'https://example.com/youtube.com'
} should match snapshot 1`] = `
<div>
  <div
    class="bg-black rounded-xl border border-white/10 flex cursor-pointer  flex-col transition-all duration-300 hover:scale-105"
  >
    <img
      alt="Test"
      class="w-full h-fullobject-cover rounded-t-xl"
      data-nimg="1"
      decoding="async"
      height="150"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fexample.com%2Fthumbnail.jpg&w=384&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fthumbnail.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fthumbnail.jpg&w=384&q=75 2x"
      style="color: transparent;"
      width="150"
    />
    <div
      class="p-2 h-20 overflow-auto"
    >
      <h2
        class="text-white text-sm font-bold"
      >
        Test
      </h2>
      <p
        class="text-white/50 text-sm"
      >
        Test
      </p>
      <p
        class="text-white/50 text-sm"
      >
        Test
      </p>
    </div>
  </div>
</div>
`;

exports[`Spinner {
  id: '1',
  name: 'longtextlongtextlongtextlongtextlongtextlongtext',
  ingredients: [],
  category: 'Test',
  area: 'Test',
  instructions: 'Test',
  thumbnail: 'https://example.com/thumbnail.jpg',
  tags: [],
  youtube: 'https://example.com/youtube.com'
} should match snapshot 1`] = `
<div>
  <div
    class="bg-black rounded-xl border border-white/10 flex cursor-pointer  flex-col transition-all duration-300 hover:scale-105"
  >
    <img
      alt="longtextlongtextlongtextlongtextlongtextlongtext"
      class="w-full h-fullobject-cover rounded-t-xl"
      data-nimg="1"
      decoding="async"
      height="150"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fexample.com%2Fthumbnail.jpg&w=384&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fthumbnail.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fthumbnail.jpg&w=384&q=75 2x"
      style="color: transparent;"
      width="150"
    />
    <div
      class="p-2 h-20 overflow-auto"
    >
      <h2
        class="text-white text-sm font-bold"
      >
        longtextlongtextlongtextlongtextlongtextlongtext
      </h2>
      <p
        class="text-white/50 text-sm"
      >
        Test
      </p>
      <p
        class="text-white/50 text-sm"
      >
        Test
      </p>
    </div>
  </div>
</div>
`;
